utils/crypto/bls/signer/localsigner/localsigner.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package localsigner
+
+import (
+	"crypto/rand"
+	"errors"
+	"runtime"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+
+	blst "github.com/supranational/blst/bindings/go"
+)
+
+var (
+	ErrFailedSecretKeyDeserialize            = errors.New("couldn't deserialize secret key")
+	_                             bls.Signer = (*LocalSigner)(nil)
+)
+
+type secretKey = blst.SecretKey
+
+type LocalSigner struct {
+	sk *secretKey
+}
+
+// NewSecretKey generates a new secret key from the local source of
+// cryptographically secure randomness.
+func NewSigner() (*LocalSigner, error) {
```

Comment:
`localsigner.NewSigner` kind of stutters - should this just be `New`?
```suggestion
func New() (*LocalSigner, error) {
```